src/store.ts

```diff
@@ -121,8 +138,12 @@ const store: TypedVuexStore = new TypedVuexStore({
     setOnboardingData(state: VuexStoreState, onboardingData: AppOnboardingData) {
       state.onboardingData = onboardingData;
     },
+    setOrderByNewest(state: VuexStoreState, orderByNewest: boolean) {
+      state.orderByNewest = orderByNewest;
+      state.semesters = sorted(state.semesters, orderByNewest);
+    },
     setSemesters(state: VuexStoreState, semesters: readonly FirestoreSemester[]) {
-      state.semesters = semesters;
+      state.semesters = sorted(semesters, state.orderByNewest);
```

Comment:
Won't this cause a lot of unnecessary sorting? It seems that `setOrderByNewest` already does this.